### DISCUSSION TITLE
Tearing in qemu

### DISCUSSION THREAD

**Discussion by replabrobin** [labwc/labwc#1264] [General]
I use slim+openbox in artix-runit and wanted to test labwc. I was able to set up a simple kvm and installed labwc-git, wlroots, xwayland, swaybg, bemenu and some others needed as dependencies or for building. 

I also installed some xfce4 progs eg xfce4-terminal. I set up a simple config in ~/.config/labwc as per the getting started.

running labwc works, but the screen is not very stable and I see a lot of tearing/flashing as I use the menu or move a terminal window.
I have seen a message on exit saying that 20ms response time is too slow. I suppose that could be the main cause.

The vm is using a spice display and virtio vga for the video. This sort of setup works fine for X11.

Do you know of a workaround for this problem?

I thank you for the effort and I can say this works well on real hardware.

**Comment by Consolatis** [comment]
> running labwc works, but the screen is not very stable and I see a lot of tearing/flashing as I use the menu or move a terminal window.

> The vm is using a spice display and virtio vga for the video. This sort of setup works fine for X11.

Wlroots might not play well with the qemu virtio vga drivers. You could try starting labwc with `WLR_RENDERER=pixman`, that should cause the software renderer to be taken instead. I am however not sure if the actual render part is the culprit here  or if its the display part.
https://gitlab.freedesktop.org/wlroots/wlroots/-/blob/master/docs/env_vars.md lists some more env vars that could be tested, especially `WLR_DRM_NO_ATOMIC` and `WLR_DRM_NO_MODIFIERS` / `WLR_EGL_NO_MODIFIERS`.

> I have seen a message on exit saying that 20ms response time is too slow. I suppose that could be the main cause.

That message is from libinput and can usually be ignored.

**Reply by MrBitBucket** [reply]
Thanks re libinput. I tried a couple of things related to the host qemu video and also the labwc renderer in the guest. Nothing really seemed to improve the situation. 

I noticed the same issue in waybox.

**Reply by replabrobin** [reply]
Sorry for responding with a different id; comes from having too many accounts :(
